src/testing/flow_solver_cases/periodic_1D_flow.cpp

```diff
@@ -0,0 +1,125 @@
+Periodic1DFlow<dim, nstate>::Periodic1DFlow(const PHiLiP::Parameters::AllParameters *const parameters_input)
+        : FlowSolverCaseBase<dim, nstate>(parameters_input)
+        , number_of_refinements(this->all_param.grid_refinement_study_param.num_refinements)
+        , domain_left(this->all_param.grid_refinement_study_param.grid_left)
+        , domain_right(this->all_param.grid_refinement_study_param.grid_right)
+        , unsteady_data_table_filename_with_extension(parameters_input->flow_solver_param.unsteady_data_table_filename+".txt")
+{
+    // Get the flow case type -- I think this was for assigning the TGV stuff
+    //using FlowCaseEnum = Parameters::FlowSolverParam::FlowCaseType;
+    //const FlowCaseEnum flow_type = parameters_input->flow_solver_param.flow_case_type;
+
+    //this->number_times_refined_by_half = 0;
+}
+
+template <int dim, int nstate>
+void Periodic1DFlow<dim,nstate>::display_flow_solver_setup(std::shared_ptr<InitialConditionFunction<dim,nstate,double>> /*initial_condition*/) const
+{
+    using PDE_enum = Parameters::AllParameters::PartialDifferentialEquation;
+    const PDE_enum pde_type = this->all_param.pde_type;
+    std::string pde_string;
+    if (pde_type == PDE_enum::advection)                {pde_string = "advection";}
+    this->pcout << "- PDE Type: " << pde_string << std::endl;
+    this->pcout << "- Polynomial degree: " << this->all_param.grid_refinement_study_param.poly_degree << std::endl;
+    this->pcout << "- Constant time step size: " << this->all_param.ode_solver_param.initial_time_step << std::endl;
+    this->pcout << "- Final time: " << this->all_param.flow_solver_param.final_time << std::endl;
+
+}
+
+template <int dim, int nstate>
+std::shared_ptr<Triangulation> Periodic1DFlow<dim,nstate>::generate_grid() const
+{
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (
+#if PHILIP_DIM!=1
+            this->mpi_communicator
+#endif
+    );
+    
+    dealii::GridGenerator::hyper_cube(*grid, domain_left, domain_right, true); //colorize = true
+    std::vector<dealii::GridTools::PeriodicFacePair<typename Triangulation::cell_iterator> > matched_pairs;
+    dealii::GridTools::collect_periodic_faces(*grid, 0,1,0, matched_pairs);
+    grid->add_periodicity(matched_pairs);
+    grid->refine_global(number_of_refinements);
+
+    // Display the information about the grid
+    this->pcout << "\n- GRID INFORMATION:" << std::endl;
+    // pcout << "- - Grid type: " << grid_type_string << std::endl;
+    this->pcout << "- - Domain dimensionality: " << dim << std::endl;
+    this->pcout << "- - Domain left: " << domain_left << std::endl;
+    this->pcout << "- - Domain right: " << domain_right << std::endl;
+    this->pcout << "- - Number of refinements: " << number_of_refinements << std::endl;
+
+    return grid;
+}
+
+template <int dim, int nstate>
+double Periodic1DFlow<dim,nstate>::get_constant_time_step(std::shared_ptr<DGBase<dim,double>> /* dg*/) const
```

Comment:
Can remove this function from the class since it is the same as the definition in the base class `flow_solver_case_base.cpp`

Reply:
Fixed in next commit.

Reply:
Please only mark as fixed after the commit has been pushed

Reply:
In the future, let the reviewers mark comments as resolved, allows us to actually check the change and approve it or further comment on it